crates/oxc_linter/src/service/runtime.rs

```diff
@@ -228,69 +629,29 @@ impl Runtime {
             .build(allocator.alloc(ret.program));
 
         if !semantic_ret.errors.is_empty() {
-            return semantic_ret.errors.into_iter().map(|err| Message::new(err, None)).collect();
+            return Err(semantic_ret.errors);
         };
 
         let mut semantic = semantic_ret.semantic;
         semantic.set_irregular_whitespaces(ret.irregular_whitespaces);
 
         let module_record = Arc::new(ModuleRecord::new(path, &ret.module_record, &semantic));
 
+        let mut requested_module_paths: Vec<(CompactStr, Arc<OsStr>)> = vec![];
+
         // If import plugin is enabled.
-        if self.resolver.is_some() {
-            self.modules.add_resolved_module(path, Arc::clone(&module_record));
+        if let Some(resolver) = &self.resolver {
             // Retrieve all dependent modules from this module.
             let dir = path.parent().unwrap();
-            module_record
+            requested_module_paths = module_record
                 .requested_modules
-                .keys()
-                .par_bridge()
-                .map_with(self.resolver.as_ref().unwrap(), |resolver, specifier| {
-                    resolver.resolve(dir, specifier).ok().map(|r| (specifier, r))
+                .par_iter()
```

Comment:
`par_iter` is probably not required anymore, since there's no further processing other than `resolver.resolve`. Spawning a new thread here may be more expensive than `resolver.resolve` it self. We need the threads for heavy duties such as parse + lint.

Reply:
resolved in c074e3b